src/main/java/org/jboss/logmanager/handlers/DelayedHandler.java

```diff
@@ -0,0 +1,228 @@
+                // Always attempt to drain the queue
+                ExtLogRecord record;
+                while ((record = logRecords.pollFirst()) != null) {
+                    StandardOutputStreams.printError(formatter.format(record));
+                }
+            }
+        }
+        activated = false;
+        super.close();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Note that once this is invoked the handler will be activated and the messages will no longer be queued. If more
+     * than one child handler is required the {@link #setHandlers(Handler[])} should be used.
+     * </p>
+     *
+     * @see #setHandlers(Handler[])
+     */
+    @Override
+    public void addHandler(final Handler handler) throws SecurityException {
+        super.addHandler(handler);
+        activate();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Note that once this is invoked the handler will be activated and the messages will no longer be queued.
+     * </p>
+     */
+    @Override
+    public Handler[] setHandlers(final Handler[] newHandlers) throws SecurityException {
+        final Handler[] result = super.setHandlers(newHandlers);
+        activate();
+        return result;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Note that if the last child handler is removed the handler will no longer be activated and the messages will
+     * again be queued.
+     * </p>
+     *
+     * @see #clearHandlers()
+     */
+    @Override
+    public void removeHandler(final Handler handler) throws SecurityException {
+        super.removeHandler(handler);
+        activated = handlers.length != 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Note that once this is invoked the handler will no longer be activated and messages will again be queued.
+     * </p>
+     *
+     * @see #removeHandler(Handler)
+     */
+    @Override
+    public Handler[] clearHandlers() throws SecurityException {
+        activated = false;
+        return super.clearHandlers();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This can be overridden to always require the caller calculation by setting the
+     * {@link #setCallerCalculationRequired(boolean)} value to {@code true} or permanently disabled for this handler by
+     * setting the value to {@code false}.
+     * </p>
+     *
+     * @see #setCallerCalculationRequired(boolean)
+     */
+    @Override
+    public boolean isCallerCalculationRequired() {
+        final int callerCalculationState = this.callerCalculationState;
+        if (callerCalculationState == 0) {
+            return false;
+        } else if (callerCalculationState == 1) {
+            return true;
+        }
+        // It's possible being a delayed handler that this could report a false negative. However since the this can
+        // manually be overridden it's best to assume we don't want the caller information.
+        return super.isCallerCalculationRequired();
```

Comment:
Change this whole method to:
`return forceCallerCalculation || super.isCallerCalculationRequired();`

Reply:
IOW we never want to force _not_ calculating the caller, but we also don't want to calculate the caller unless we must.

Reply:
Okay fair point. I guess that really does make more sense in a case where the formatter may really want the caller information, but the user made a mistake of setting it to `false`.